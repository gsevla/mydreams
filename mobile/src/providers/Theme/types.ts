export type ThemeBase = {
  sizes: {
    border: {
      thin: number;
      large: number;
    };
    radius: {
      small: number;
      medium: number;
    };
    input: {
      height: number | string;
      minWidth: number | string;
      padding: number;
    };
    button: {
      height: number | string;
      minWidth: number | string;
      paddingVertical: number;
      paddingHorizontal: number;
    };
    text: {
      h1: number;
      h2: number;
      h3: number;
      h4: number;
      h5: number;
      h6: number;
      h7: number;
      h8: number;
      h9: number;
    };
  };
};

export type Theme = ThemeBase & {
  dark: boolean;
  colors: {
    primary: string;
    secondary: string;
    background: string;
    overlay: string;
    text: string;
    textLight: string;
    border: string;
    error: string;
  };
};
