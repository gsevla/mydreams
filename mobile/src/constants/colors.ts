export const Colors = {
  white: "#FFFFFF",
  black: "#000000",
  revell: "#eeeeee",
  black60: "#00000099",
  gunmetal: "#242E38",
  meatBrow: "#EAC43D",
  cyanCobaltBlue: "#2A5C99",
  turquoiseSurf: "#14C0CC",
  androidGreen: "#A6CD4E",
  lightGray: "#D3D3D3",
  redOrange: "#FF3333",
};
