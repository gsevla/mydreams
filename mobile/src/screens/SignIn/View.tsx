import React from "react";
import { KeyboardAvoidingView, ScrollView, View } from "react-native";
import { Text } from "@components/Text";
import { TextInput } from "@components/TextInput";
import { SizedBox } from "@components/SizedBox";
import { ButtonText } from "@components/Button/Text";
import { ButtonSolid } from "@components/Button/Solid";
import { SignInScreenViewProps } from "./types";

export function SignInScreenView({
  styles,
  theme,
  navigateToSignUpScreen,
  authenticate,
}: SignInScreenViewProps) {
  return (
    <KeyboardAvoidingView style={styles.container} behavior="padding">
      <View style={styles.headerContainer}>
        <Text fontSize="h1" color={theme.colors.primary}>
          MyDreams
        </Text>
        <SizedBox height={8} />
        <Text fontSize="h1" color={theme.colors.secondary}>
          O lugar ideal para guardar seus sonhos!
        </Text>
      </View>
      <View style={styles.form}>
        <TextInput
          value=""
          label="Usuário"
          placeholder="gabrielalves"
          onChangeText={() => {}}
        />
        <SizedBox height={24} />
        <TextInput
          value=""
          label="Senha"
          placeholder="********"
          onChangeText={() => {}}
          isPassword
        />
      </View>
      <View style={styles.buttonsContainer}>
        <ButtonSolid label="Entrar" onPress={authenticate} />
        <SizedBox height={8} />
        <ButtonText
          label="Não possuo conta!"
          onPress={navigateToSignUpScreen}
        />
      </View>
    </KeyboardAvoidingView>
  );
}
