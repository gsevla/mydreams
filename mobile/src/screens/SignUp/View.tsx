import React from "react";
import { ButtonSolid } from "@components/Button/Solid";
import { Select } from "@components/Select";
import { SizedBox } from "@components/SizedBox";
import { TextInput } from "@components/TextInput";
import { KeyboardAvoidingView, View } from "react-native";

export function SignUpScreenView({ styles, signUp }) {
  return (
    <View style={styles.container}>
      <View style={styles.form}>
        <TextInput
          value=""
          label="Usuário"
          placeholder="gabrielalves"
          onChangeText={() => {}}
        />
        <SizedBox height={24} />
        <TextInput
          value=""
          label="Senha"
          placeholder="********"
          onChangeText={() => {}}
          isPassword
        />
        <SizedBox height={24} />
        <TextInput
          value=""
          label="Repita sua senha"
          placeholder="********"
          onChangeText={() => {}}
          isPassword
        />
        <SizedBox height={24} />
        <Select
          value={null}
          data={[
            {
              value: "MALE",
              label: "Masculino",
            },
            {
              value: "FEMALE",
              label: "Feminino",
            },
            {
              value: "OTHER",
              label: "Outro",
            },
          ]}
          label="Gênero"
          placeholder="Selecione seu gênero"
          onSelectOption={(option) => {
            console.log("bbbb", option);
          }}
        />
      </View>
      <View
        pointerEvents="box-none"
        style={[styles.buttonsContainer, { backgroundColor: "transparent" }]}
      >
        <ButtonSolid label="Cadastrar" onPress={signUp} />
      </View>
    </View>
  );
}
