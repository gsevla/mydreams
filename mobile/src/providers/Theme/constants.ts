import { Colors } from "@constants/colors";
import type { ThemeBase, Theme } from "./types";

const Base: ThemeBase = {
  sizes: {
    border: {
      thin: 1,
      large: 2,
    },
    button: {
      height: 48,
      minWidth: 124,
      paddingVertical: 8,
      paddingHorizontal: 16,
    },
    input: {
      height: 48,
      minWidth: "100%",
      padding: 8,
    },
    radius: {
      small: 4,
      medium: 8,
    },
    text: {
      h1: 24,
      h2: 22,
      h3: 20,
      h4: 18,
      h5: 16,
      h6: 14,
      h7: 12,
      h8: 10,
      h9: 8,
    },
  },
};

export const Light: Theme = {
  ...Base,
  dark: false,
  colors: {
    background: Colors.revell,
    overlay: Colors.black60,
    primary: Colors.cyanCobaltBlue,
    secondary: Colors.turquoiseSurf,
    text: Colors.gunmetal,
    textLight: Colors.white,
    border: Colors.gunmetal,
    error: Colors.redOrange,
  },
};
