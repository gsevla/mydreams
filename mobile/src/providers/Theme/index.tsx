import React, { createContext } from "react";
import { Theme } from "./types";
import { Light } from "./constants";

type Props = { children: React.ReactNode };

export const ThemeContext = createContext({} as { theme: Theme });

export function ThemeProvider({ children }: Props) {
  return (
    <ThemeContext.Provider value={{ theme: Light }}>
      {children}
    </ThemeContext.Provider>
  );
}
