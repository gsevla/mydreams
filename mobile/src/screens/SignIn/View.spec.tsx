import React from "react";
import { Light } from "@providers/Theme/constants";
import { withProviders } from "@test/utils/withProviders";
import { render } from "@testing-library/react-native";
import { SignInScreenView, SignInScreenViewProps } from "./View";

const createProps = (props: Partial<SignInScreenViewProps> = {}) =>
  ({
    authenticate: jest.fn(),
    navigateToSignUpScreen: jest.fn(),
    setEmail: jest.fn(),
    setPassword: jest.fn(),
    theme: Light,
    ...props,
  } as SignInScreenViewProps);

describe("SignInScreenView Component", () => {
  it("should match snapshot", () => {
    const { toJSON } = render(
      withProviders(<SignInScreenView {...createProps()} />)
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
