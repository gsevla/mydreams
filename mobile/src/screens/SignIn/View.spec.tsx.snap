// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInScreenView Component should match snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    [
      {
        "flex": 1,
        "justifyContent": "center",
        "paddingVertical": 24,
      },
      {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessibilityRole="text"
      style={
        {
          "color": "#2A5C99",
          "fontSize": 24,
          "textAlign": "auto",
        }
      }
    >
      MyDreams
    </Text>
    <View
      accessible={false}
      style={
        {
          "height": 8,
          "width": undefined,
        }
      }
    />
    <Text
      accessibilityRole="text"
      style={
        {
          "color": "#14C0CC",
          "fontSize": 24,
          "textAlign": "auto",
        }
      }
    >
      O lugar ideal para guardar seus sonhos!
    </Text>
  </View>
  <View
    style={
      {
        "flexGrow": 0,
      }
    }
  >
    <View>
      <Text
        accessibilityRole="text"
        style={
          {
            "color": "#242E38",
            "fontSize": 16,
            "textAlign": "auto",
          }
        }
      >
        Email
      </Text>
      <View
        accessible={false}
        style={
          {
            "height": 8,
            "width": undefined,
          }
        }
      />
      <View
        style={
          {
            "justifyContent": "center",
          }
        }
      >
        <TextInput
          autoCapitalize="none"
          autoCorrect={false}
          keyboardType="default"
          onChangeText={[MockFunction]}
          placeholder="gabriel@alves.com"
          returnKeyType="default"
          secureTextEntry={false}
          style={
            [
              {
                "borderColor": "#242E38",
                "borderRadius": 4,
                "borderWidth": 1,
                "paddingRight": 32,
              },
              {
                "height": 48,
                "minWidth": "100%",
                "padding": 8,
              },
            ]
          }
        />
      </View>
      <View
        accessible={false}
        style={
          {
            "height": 8,
            "width": undefined,
          }
        }
      />
      <View
        style={
          {
            "paddingLeft": 8,
          }
        }
      >
        <Text
          accessibilityRole="text"
          style={
            {
              "color": "#FF3333",
              "fontSize": 16,
              "textAlign": "auto",
            }
          }
        />
      </View>
    </View>
    <View
      accessible={false}
      style={
        {
          "height": 24,
          "width": undefined,
        }
      }
    />
    <View>
      <Text
        accessibilityRole="text"
        style={
          {
            "color": "#242E38",
            "fontSize": 16,
            "textAlign": "auto",
          }
        }
      >
        Senha
      </Text>
      <View
        accessible={false}
        style={
          {
            "height": 8,
            "width": undefined,
          }
        }
      />
      <View
        style={
          {
            "justifyContent": "center",
          }
        }
      >
        <TextInput
          autoCapitalize="none"
          autoCorrect={false}
          keyboardType="default"
          onChangeText={[MockFunction]}
          placeholder="********"
          returnKeyType="default"
          secureTextEntry={true}
          style={
            [
              {
                "borderColor": "#242E38",
                "borderRadius": 4,
                "borderWidth": 1,
                "paddingRight": 32,
              },
              {
                "height": 48,
                "minWidth": "100%",
                "padding": 8,
              },
            ]
          }
        />
        <Text />
      </View>
      <View
        accessible={false}
        style={
          {
            "height": 8,
            "width": undefined,
          }
        }
      />
      <View
        style={
          {
            "paddingLeft": 8,
          }
        }
      >
        <Text
          accessibilityRole="text"
          style={
            {
              "color": "#FF3333",
              "fontSize": 16,
              "textAlign": "auto",
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "flex-end",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#2A5C99",
          "borderRadius": 8,
          "height": 48,
          "justifyContent": "center",
          "minWidth": 124,
          "opacity": 1,
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        }
      }
    >
      <Text
        accessibilityRole="text"
        style={
          {
            "color": "#FFFFFF",
            "fontSize": 18,
            "textAlign": "auto",
          }
        }
      >
        Entrar
      </Text>
    </View>
    <View
      accessible={false}
      style={
        {
          "height": 8,
          "width": undefined,
        }
      }
    />
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderRadius": 8,
          "height": 48,
          "justifyContent": "center",
          "minWidth": 124,
          "opacity": 1,
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        }
      }
    >
      <Text
        accessibilityRole="text"
        style={
          {
            "color": "#2A5C99",
            "fontSize": 18,
            "textAlign": "auto",
          }
        }
      >
        NÃ£o possuo conta!
      </Text>
    </View>
  </View>
</View>
`;
